internal/cli/opsmanager/clusters/describe.go

```diff
@@ -40,20 +40,34 @@ func (opts *DescribeOpts) initStore() error {
 	return err
 }
 
-func (opts *DescribeOpts) Run() error {
-	result, err := opts.store.GetAutomationConfig(opts.ConfigProjectID())
+var describeTemplate = `ID	NAME	TYPE
+{{.ID}}	{{.ClusterName}}	{{.TypeName}}
+`
 
+func (opts *DescribeOpts) Run() error {
+	r, err := opts.cluster()
 	if err != nil {
 		return err
 	}
 
-	clusterConfigs := convert.FromAutomationConfig(result)
-	for _, rs := range clusterConfigs {
+	return output.Print(config.Default(), describeTemplate, r)
+}
+
+func (opts *DescribeOpts) cluster() (interface{}, error) {
+	if config.Output() == "" {
```

Comment:
Seems like the template is only valid with this Cluster type & OM. Will there be a template for the output mode in L67 / the Atlas cluster type?

I'm wondering if there's a way we can avoid using interface{} here and in the other command, but I'm not sure how easy that'd be with the changes to the store

Reply:
if no output is selected we call the clusters api, and use the template
if we specify json we use the automation API and transform it to the cli format

The original implementation of this using the automation config was so that listing or describing would let you take that same config modify it and apply it again, something you can't do using the clusters API

That approach had a downside we had no cluster ID, this was a problem if you needed access to the metrics commands so we this change I'm making sure users get back that missing info.

This is my long way on why of this changes

>I'm wondering if there's a way we can avoid using interface{}

Sadly not with go :(

Reply:
Makes sense -- perhaps this is the best way for now then